lib/rest-api.php

```diff
@@ -183,3 +183,37 @@ function gutenberg_auto_draft_get_sample_permalink( $permalink, $id, $title, $na
 	return $permalink;
 }
 add_filter( 'get_sample_permalink', 'gutenberg_auto_draft_get_sample_permalink', 10, 5 );
+
+/**
+ * Filters WP_User_Query arguments when querying users via the REST API.
+ *
+ * Allow who=has_published_post.
+ *
+ * @param array           $prepared_args Array of arguments for WP_User_Query.
+ * @param WP_REST_Request $request       The REST API request.
+ *
+ * @return array Returns modified $prepared_args.
+ */
+function gutenberg_rest_user_query_has_published_posts( $prepared_args, $request ) {
+	if ( ! empty( $request['who'] ) && 'has_published_posts' === $request['who'] ) {
+		$prepared_args['has_published_posts'] = get_post_types( array( 'show_in_rest' => true ), 'names' );
```

Comment:
What would be the different from using `$prepared_args['has_published_posts']=true`? Upon [reading the docs](https://developer.wordpress.org/reference/classes/wp_user_query/#has-published-posts-parameter), I'm not sure whether there's post types that aren't shown in rest requests but can be public.

Reply:
I don't think it will make any difference, and I'd probably just use `true` as well.
The reason I did it using `get_post_types( array( 'show_in_rest' => true ), 'names' );` is that I was thinking of making it easier to backport this change to core later on. So this is just for consistency with https://github.com/WordPress/wordpress-develop/blob/5cd6bfe2e0f35922c70b4243288d49f239a09ab4/src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php#L301 and backporting would simply require changing the condition above that line (and also add `has_published_posts` to [this array](https://github.com/WordPress/wordpress-develop/blob/5cd6bfe2e0f35922c70b4243288d49f239a09ab4/src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php#L1556-L1558)).
We can change it to `true`, and I can add a note in the function comments that this should also be changed in core when backporting if you think that's best :+1:

Reply:
Nice, it makes sense leaving it this way then.